cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -1039,7 +1039,15 @@ public void batchNextRunTimes(FullHttpRequest request,
   private List<BatchProgramSchedule> batchRunTimes(String namespace, Collection<? extends BatchProgram> programs,
                                                    boolean previous) throws Exception {
     List<ProgramId> programIds = programs.stream()
-      .map(p -> new ProgramId(namespace, p.getAppId(), p.getProgramType(), p.getProgramId()))
+      .map(p -> {
+        try {
+          return new ProgramId(namespace, p.getAppId(), getLatestAppVersion(new NamespaceId(namespace), p.getAppId()),
+                               p.getProgramType(), p.getProgramId());
+        } catch (ApplicationNotFoundException e) {
+          // do nothing
```

Comment:
plz comment on my code style, my java is bad @sanjanasandeep @de-lan @jingtaowu3

Reply:
in line 1068, we need to put a not found exception in the response, so we can't ignore it here.